src/App.tsx

```diff
@@ -2,84 +2,128 @@ import React from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
 
-export const goods = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+type Props = {};
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  goods: string[],
+  chosenElem: string,
+  clearBtn: boolean,
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<Props, State> {
+  state = {
+    goods: [
+      'Dumplings',
+      'Carrot',
+      'Eggs',
+      'Ice cream',
+      'Apple',
+      'Bread',
+      'Fish',
+      'Honey',
+      'Jam',
+      'Garlic',
+    ],
+    chosenElem: 'Jam',
+    clearBtn: true,
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  chosenGood = (good: string) => {
+    this.setState({ chosenElem: good });
+    this.setState({ clearBtn: true });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  clear = () => {
+    this.setState({ chosenElem: 'No' });
+    this.setState({ clearBtn: false });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render(): React.ReactNode {
+    const { goods } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <h1 className="title is-flex is-align-items-center">
+          { this.state.chosenElem === 'No'
+            ? 'No goods selected'
+            : `${this.state.chosenElem} is selected` }
+          {this.state.clearBtn
+            ? (
+              <div>
+                <button
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  onClick={() => {
+                    this.clear();
+                  }}
+                >
+                  {null}
+                </button>
+              </div>
```

Comment:
```suggestion
                <button
                  data-cy="ClearButton"
                  type="button"
                  className="delete ml-3"
                  onClick={() => {
                    this.clear();
                  }}
                 />
```

`div` is redundant, also you can remove `{null} `value from button